retrofit/src/main/java/retrofit/RxSupport.java

```diff
@@ -21,24 +24,35 @@
   RxSupport() {
   }
 
-  Observable createRequestObservable(final Invoker invoker) {
-    return Observable.create(new Observable.OnSubscribe<Object>() {
-      @Override public void call(final Subscriber<? super Object> subscriber) {
-        invoker.invoke(new Invoker.Callback() {
-          @Override public void next(Object o) {
-            if (!subscriber.isUnsubscribed()) {
-              subscriber.onNext(o);
-              subscriber.onCompleted();
-            }
-          }
+  Observable createRequestObservable(final Invoker invoker, final RequestBuilder requestBuilder,
+      final RequestInterceptor requestInterceptor) {
+    return Observable.defer(new Func0<Observable<Object>>() {
```

Comment:
Any reason you chose to use `defer` here? I'm pretty sure we can just put the `requestInterceptor` line first in the `OnSubscribe`'s `call` method to achieve the same effect without allocating the wrapper.